op-node/rollup/derive/l1_block_info.go

```diff
@@ -115,7 +119,7 @@ func L1InfoDepositTxData(data []byte) (L1BlockInfo, error) {
 
 // L1InfoDeposit creates a L1 Info deposit transaction based on the L1 block,
 // and the L2 block-height difference with the start of the epoch.
-func L1InfoDeposit(seqNumber uint64, block eth.BlockInfo, sysCfg eth.SystemConfig) (*types.DepositTx, error) {
+func L1InfoDeposit(seqNumber uint64, block eth.BlockInfo, sysCfg eth.SystemConfig, regolith bool) (*types.DepositTx, error) {
```

Comment:
Perhaps it'd be better to pass the config in directly, in case we need this method to support additional HFs in the future? I've seen this done in Geth in various places.

This comment isn't blocking - if you prefer the way it's done here, I'm happy to defer to your judgement.

Reply:
Agree on this, less refactoring in the future

Reply:
Looks like it would add some overhead to the tests

Reply:
Adding a full config for a single boolean does add a lot of overhead, which I'd like to avoid. This code-path is meant to be simple, it's up to the struct-based utils like the AttributesBuilder to handle the configuration.